### PR TITLE
feat: LW-10725 add co signer fields of the current user

### PR DESCRIPTION
**Pull request by greatertomi**
# Checklist

- [x] JIRA - LW-10725
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Add one more cosigner to the cosigners on the create shared wallet flow.

### CODE REVIEW

**Review by vetalcore — APPROVED** [review]

**Comment by vetalcore** [comment]
@szymonmaslowski @greatertomi would you be able to approve the snapshots generated by chromatic? some of them seem to have sizing issues*

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/SharedWalletCreationStore.tsx [hunk 1]
```diff
@@ -272,10 +272,12 @@ const makeStateMachine = ({
   [SharedWalletCreationStep.Quorum]: (prevState, action) => {
     if (action.type === SharedWalletActionType.BACK) {
       // Having two cosigner fields fixed we need to fall back to two entries if user specified
-      // just one because the empty one was filtere out in brevious step.
+      // just one because the empty one was filters out in previous step.
       const coSigners = [
         prevState.coSigners[0] || createEmptyCosignerObject(),
         prevState.coSigners[1] || createEmptyCosignerObject(),
+        // eslint-disable-next-line no-magic-numbers
+        prevState.coSigners[2] || createEmptyCosignerObject(),
```
Comment (szymonmaslowski) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/SharedWalletCreationStore.tsx:280]: No related to this line, but we need to make the keys input of the first object disabled so the user cannot modify it. It should also have a different label. [Here is the design](https://www.figma.com/design/RPN3V4Et7hMbXkV5ZIF75i/[deprecated]-Lace-Browser---Full-view-UI-Design?node-id=42863-479308&t=K7kLEOAFepjr1nhe-0).

**Review by szymonmaslowski — CHANGES_REQUESTED** [review]
GJ! Leaving some comments.

**Review by danielmain — APPROVED** [review]

**Review by shawnbusuttil — APPROVED** [review]

**Review by szymonmaslowski — APPROVED** [review]

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx [hunk 2]
```diff
@@ -44,6 +45,17 @@ export const AddCoSigners = ({
...
+            labels={{
+              keys:
+                index === indexOfCoSignersDataOfCurrentUser
+                  ? t('sharedWallets.addSharedWallet.addCosigners.yourKeysInputLabel')
+                  : t('sharedWallets.addSharedWallet.addCosigners.coSignerKeysInputLabel'),
+              name:
+                index === indexOfCoSignersDataOfCurrentUser
+                  ? t('sharedWallets.addSharedWallet.addCosigners.yourNameInputLabel')
+                  : t('sharedWallets.addSharedWallet.addCosigners.coSignerNameInputLabel'),
+            }}
```
Comment (vetalcore) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx:58]:
```suggestion
            labels={{
              keys: t(`sharedWallets.addSharedWallet.addCosigners.${index === indexOfCoSignersDataOfCurrentUser ? 'yourKeysInputLabel' : 'coSignerKeysInputLabel'`),
              name: t(`sharedWallets.addSharedWallet.addCosigners.${index === indexOfCoSignersDataOfCurrentUser 'yourNameInputLabel' : 'coSignerNameInputLabel'`),
            }}
```

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/SharedWalletCreationStore.tsx [hunk 3]
```diff
@@ -8,10 +8,14 @@ import React, {
   useMemo,
   useReducer,
 } from 'react';
-import { v1 as uuid } from 'uuid';
 import { makeInitialStateProvider } from '../../initial-state-provider';
 import { StateType, TransitionHandler, defineStateShape } from '../../state-utils';
 import { CoSigner, CoSignerDirty, CoSignerError } from './AddCoSigners';
+import {
+  createCosignerObject,
```
Comment (vetalcore) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/SharedWalletCreationStore.tsx:15]: could we please follow single approach, eg:
```suggestion
  createCoSignerObject,
```

**Review by vetalcore — APPROVED** [review]

**Pull request merged**